std/range/package.d

```diff
@@ -9594,6 +9594,324 @@ auto refRange(R)(R* range)
     foo(r);
 }
 
+/**
+Bitwise adapter over Integral type Ranges. Consumes the range elements bit by bit.
+*/
+struct Bitwise(R)
+    if (isInputRange!R && isIntegral!(ElementType!R))
+{
+    alias ER = ElementType!R;
+    alias UER = Unsigned!ER;
+
+    private R parent;
+    private UER mask;
+    private int maskPos;
+    private ER accumulator;
+
+    private enum UER bitsNum = ER.sizeof * 8;
+
+    this()(auto ref R range)
+    {
+        parent = range;
+
+        static if (hasLength!R)
+        {
+            len = parent.length * bitsNum;
+        }
+
+        initPrivates();
+
+        static if (isBidirectionalRange!R)
+        {
+            initBackPrivates();
+        }
+    }
+
+    bool empty() const
```

Comment:
Unless something has changed recently, `const` is not inferred. So, if you want it to be `const` or not based on the template argument, then you have to use introspection and `static if`. That being said, it still shouldn't just be marked as `const`, because then it won't work with many ranges.